src/components/RoundForm.vue

```diff
@@ -0,0 +1,264 @@
+      min="1" max="18" size="35" placeholder="# holes" 
+      @change="changeNumHoles(courseIndex)"><br><br>
+      
+      <table border="1" cellpadding="4">
+        <tr>
+          <th>Hole</th>
+          <th v-for="num in course.numHoles" :key="num">{{ num }}</th>
+        </tr>
+        <tr v-for="(score, scoreIndex) in course.scores"  :key="score.player">
+          <td>
+            {{ score.player }}
+            <button @click="removeScore(scoreIndex, courseIndex)">
+              Remove
+            </button>
+          </td>
+          <td v-for="num in course.numHoles" :key="num">
+            <input v-model.number="score.strokes[num - 1]" number 
+            type="number" style="width: 35px;" min="0" max="20">
+          </td>
+        </tr>
+        <tr v-show="canAddScores">
+          <td>
+            <select v-model="newPlayer">
+              <option value="" default selected disabled>Select player</option>
+              <option v-for="player in newPlayerOptions" :key="player">
+                {{ player }}
+              </option>
+            </select>
+            <button @click="addScore" :disabled="newPlayer == ''">Add</button>
+          </td>
+        </tr>
+      </table><br><br>
+    </div>
+
+    <!-- wrap the button in a div so that tooltip can be showed
+    when button is disabled -->
+    <div v-tooltip="submitWarningMsg">
+      <button v-on:click="submit"
+      :disabled="submitWarningMsg != ''">
+        Submit
+      </button>
+    </div>
+
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'RoundForm',
+  data: function() {
+    return {
+      msg: 'Getting input data ...',
+      date: '',
+      club: {},
+      courses: [],
+      newCourse: {},
+      newPlayer: '',
+      clubOptions: [],
+      courseOptions: [],
+      playerOptions: [],
+    }
+  },
+  computed: {
+    newCourseOptions () {
+      let existingCourses = this.courses.map(c => c.name)
+      return this.courseOptions.filter(c => !existingCourses.includes(c.name))
+    },
+    canAddCourses () {
+      return this.courses.length < 2
+    },
+    newPlayerOptions () {
+      if (this.courses.length == 0) {
+        return []
+      }
+      let players = this.courses[0].scores.map(s => s.player)
+      return this.playerOptions.filter(p => !players.includes(p))
+    },
+    canAddScores () {
+      return this.courses[0].scores.length < 4
+    },
+    submitWarningMsg () {
+      if (!this.date) return 'Missing date'
+      if (Object.keys(this.club).length == 0) return 'Missing club'
+      if (this.courses.length == 0)  return 'Missing course'
+      if (!this.courses.every(c => c.tees)) return 'Missing tees'
+      if (!this.courses.every(c => c.numHoles > 0)) return 'Missing holes'
+      if (!this.courses.every(c => c.scores.length > 0)) {
+        return 'Missing players'
+      }
+      let strokes = this.courses.flatMap(c => c.scores.flatMap(s => s.strokes))
```

Comment:
Love being able to use a good ole' `flatMap` when I can

Reply:
🔥